portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -65,7 +53,38 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
 
       DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
       datastore.put(comment);
+      int max = getRequestNum(request);  
+      response.sendRedirect("/index.html?limit=" + max);
+  }
 
-      response.sendRedirect("/index.html");
+  private int getRequestNum(HttpServletRequest request) {
+      String limitString = request.getParameter("limit");
+      int limit;
+      try {
+          limit = Integer.parseInt(limitString);
+      } catch(NumberFormatException e) {
+          return 10;
+      }
+      return limit;
+  } 
+
+  private ArrayList<Comment> getComments(int limit) {
+    Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery res = datastore.prepare(query);
+
+    Iterator<Entity> iter = res.asIterable().iterator();
+    ArrayList<Comment> comments = new ArrayList<Comment>();
+    int i = 0;
+    while(iter.hasNext() && i < limit) {
```

Comment:
This is subjective, but I think code is easier to read when there are fewer variables, and each variable has as limited a scope as possible. 

I think one way this can be done is something like this - what do you think?

```java
for (Entity entity : res.asIterable()) {
  // code that creates a comment
  comments.add(comment);
  if (comments.size() > limit) {
    break;
  }
}
```
This eliminates the iter, i, and curComment variables.